### PR TITLE
fontconfig: add pkgconf build requirement + libgettext on Macos

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **fontconfig/all**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SpaceIm** [comment]
I have also link errors on Macos (in gdal when podofo is enabled). Seems that fontconfig depends on gettext on macos: https://github.com/conan-community/community/issues/307

File: recipes/fontconfig/all/conanfile.py [hunk 1]
```diff
@@ -40,9 +40,12 @@ def requirements(self):
         self.requires("expat/2.2.10")
         if self.settings.os == "Linux":
             self.requires("libuuid/1.0.3")
+        elif self.settings.os == "Macos":
+            self.requires("libgettext/0.20.1")
```
Comment (SpaceIm) [inline recipes/fontconfig/all/conanfile.py:44]: I'm not sure, but I suspect that it's required if not Linux.

**Comment by SpaceIm** [comment]
@ulirianies some CI issue? It's running for 5 hours.

**Review by SSE4 — APPROVED** [review]

File: recipes/fontconfig/all/conanfile.py (hunk 1 above)
Comment (ericLemanissier) [reply recipes/fontconfig/all/conanfile.py:44]: do we have an idea why the test recipe passes on C3I without this ? (the idea would be to improve the test recipe)

Comment (SSE4) [reply recipes/fontconfig/all/conanfile.py:44]: that might be gettext installed from brew (explicitly or as dependency)

**Review by uilianries — APPROVED** [review]

**Review by ericLemanissier — APPROVED** [review]

**Comment by uilianries** [comment]
@SpaceIm Another PR is consuming a lot of resources, delaying other PRs.

**Pull request merged**